test/extensions/filters/http/common/fuzz/uber_filter.cc

```diff
@@ -38,6 +38,16 @@ UberFilterFuzzer::UberFilterFuzzer() : async_request_{&cluster_manager_.async_cl
   ON_CALL(filter_callback_, addAccessLogHandler(_))
       .WillByDefault(
           Invoke([&](AccessLog::InstanceSharedPtr handler) -> void { access_logger_ = handler; }));
+  // This handles stopping execution after a direct response is sent.
+  ON_CALL(decoder_callbacks_, sendLocalReply(_, _, _, _, _))
+      .WillByDefault(
+          Invoke([this](Http::Code code, absl::string_view body,
+                        std::function<void(Http::ResponseHeaderMap & headers)> modify_headers,
+                        const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
+                        absl::string_view details) {
+            enabled_ = false;
```

Comment:
personally I just wanted to do `decoder_filter_.reset()` and return early when `!filter`, but for some reason that wouldn't actually reflect in the method that does the decoding, `runData`.